1-js-and-async-programming/2-asynchronous-js/3-async-await.js

```diff
@@ -17,8 +17,40 @@
  *
  * @returns {Promise<string>} Logs the subscription name as a string.
  */
+const { getUsers, getLikedMovies, getDislikedMovies,  getUserSubscriptionByUserId} = require('./utils/mocked-api');
+const getMovieCount = (movieList, userId) => {
+  const userMovies = movieList.find(item => item.userId === userId);
+  return userMovies ? userMovies.movies.length : 0;
+}
+const getReviewerSubscriptions = async (users, likedMovies, dislikedMovies) => {
+    const reviewerUsers = await Promise.all(
+      users.map(async user => {
+        const likesCount = getMovieCount(likedMovies, user.id);
+        const dislikesCount = getMovieCount(dislikedMovies, user.id);
+        if (likesCount < dislikesCount) {
+          const subscription = await getUserSubscriptionByUserId(user.id);
+          return subscription?.subscription || null;
+        }
+        return null;
+      })
+    )
+    return reviewerUsers.filter(subscription => subscription !== null)
+}
 const getCommonDislikedSubscription = async () => {
-  // Add your code here
+  try {
+    const [users, likedMovies, dislikedMovies] = await Promise.all([getUsers(), getLikedMovies(),getDislikedMovies()]);
+    const subscriptions = await getReviewerSubscriptions(users, likedMovies, dislikedMovies);
+    if(subscriptions.length === 0) return "No subscription for people with more dislikes than likes";
+    const subscriptionCounts = subscriptions.reduce((acc, sub) => {
+      acc[sub] = (acc[sub] || 0) + 1;
+      return acc;
+    },{})
+    return Object.entries(subscriptionCounts).sort((a, b) => b[1] - a[1])[0][0];
```

Comment:
Make the code readable and easy to understand for everyone
```suggestion
    const [firstSubscription] = Object.entries(subscriptionCounts).sort((a, b) => b[1] - a[1])
    const [subscriptionType] = firstSubscription
    
    return subscriptionType
```